thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -51,7 +51,7 @@
 /**
  * This class operates an IoT device that is specified by {@link #onboard(String, String, String, JSONObject)} method.
  */
-public class ThingIFAPI implements Parcelable {
+public class ThingIFAPI<AliasType extends Alias> implements Parcelable {
```

Comment:
From the official [document](https://docs.oracle.com/javase/tutorial/java/generics/types.html) about java generics: 
> By convention, type parameter names are single, uppercase letters. 

Suggest to use a single uppercase letter.